csv/result.go

```diff
@@ -1271,3 +1305,25 @@ func NewMultiResultEncoder(c ResultEncoderConfig) flux.MultiResultEncoder {
 		Encoder:   NewResultEncoder(c),
 	}
 }
+
+// bufferedCSVReader allows for unreading a single line of the csv data
+type bufferedCSVReader struct {
```

Comment:
I guess you're using this instead of the `extraLine` that was here before? I like it, this simplifies the logic a lot.

Reply:
Yep, it replaces the need for `extraLine` logic, this struct was added to make it simpler.